MIGRATION.md

```diff
@@ -138,6 +138,47 @@ Integrations that are now exported from `@sentry/node` and `@sentry/browser` (or
 - sessionTimingIntegration
 - dedupeIntegration (enabled by default, not pluggable)
 
+## Next.js SDK Changes
+
+### Removal of the `sentry` property in your Next.js options
+
+With version 8 of the Sentry Next.js SDK, the SDK will no longer support passing Next.js options with a `sentry`
+property to `withSentryConfig`. Please use the third argument of `withSentryConfig` to configure the SDK instead:
+
+```ts
+// OLD
+const nextConfig = {
+  // Your Next.js options...
+
+  sentry: {
+    // Your Sentry SDK options...
+  },
+};
+
+module.exports = withSentryConfig(nextConfig, {
+  // Your Sentry Webpack Plugin Options...
+});
+
+// NEW
+const nextConfig = {
+  // Your Next.js options...
+};
+
+module.exports = withSentryConfig(
+  nextConfig,
+  {
+    // Your Sentry Webpack Plugin Options...
+  },
+  {
+    // Your Sentry SDK options...
+  },
```

Comment:
Is there a reason we keep these two objects separately? Have you thought about moving the webpack plugin options into a sub field of the sentry SDK options? The reasons I'm bringing this up:

- To me (as an avg SDK user), I probably don't understand what the Sentry webpack plugin does, vs. the SDK. 
- In SvelteKit (granted the situation is different in the sense that we directly initialize a vite plugin), we don't have the separation but the Vite plugin options are contained in a `sourcemapsUploadOptions` field within the plugin factory function options

i don't have a super strong opinion here, so feel free to disregard but I'm not sure if the separation makes sense. wdyt?

Reply:
Yes, eventually I would like to merge the two and I am probably gonna do that within v8 even. I just wanted to get rid of the `sentry` property as a first step. This will obviously require me to update the migration guide afterwards again, but I think it is gonna be fine.

The naming is gonna be a bit tricky because in the future (and even now) the webpack plugin can do more than just source maps: release management, commits, component-name injections, ... I am likely gonna expose multiple different options that all map to the plugin's options internally somehow.